x-pack/plugins/observability_ai_assistant/server/functions/context.ts

```diff
@@ -6,49 +6,35 @@
  */
 
 import { decodeOrThrow, jsonRt } from '@kbn/io-ts-utils';
+import { Logger } from '@kbn/logging';
 import type { Serializable } from '@kbn/utility-types';
 import dedent from 'dedent';
+import { encode } from 'gpt-tokenizer';
 import * as t from 'io-ts';
 import { compact, last, omit } from 'lodash';
-import { lastValueFrom } from 'rxjs';
-import { Logger } from '@kbn/logging';
+import { lastValueFrom, Observable } from 'rxjs';
 import { FunctionRegistrationParameters } from '.';
-import { MessageRole, type Message } from '../../common/types';
+import { MessageAddEvent } from '../../common/conversation_complete';
+import { FunctionVisibility, MessageRole, type Message } from '../../common/types';
 import { concatenateChatCompletionChunks } from '../../common/utils/concatenate_chat_completion_chunks';
 import type { ObservabilityAIAssistantClient } from '../service/client';
+import { createFunctionResponseMessage } from '../service/util/create_function_response_message';
 
-export function registerRecallFunction({
+const MAX_TOKEN_COUNT_FOR_DATA_ON_SCREEN = 1000;
+
+export function registerContextFunction({
   client,
   registerFunction,
   resources,
-}: FunctionRegistrationParameters) {
+  isKnowledgeBaseAvailable,
+}: FunctionRegistrationParameters & { isKnowledgeBaseAvailable: boolean }) {
   registerFunction(
     {
-      name: 'recall',
+      name: 'context',
       contexts: ['core'],
-      description: `Use this function to recall earlier learnings. Anything you will summarize can be retrieved again later via this function.
-      
-      The learnings are sorted by score, descending.
-      
-      Make sure the query covers ONLY the following aspects:
-      - Anything you've inferred from the user's request, but is not mentioned in the user's request
-      - The functions you think might be suitable for answering the user's request. If there are multiple functions that seem suitable, create multiple queries. Use the function name in the query.  
-
-      DO NOT include the user's request. It will be added internally.
-      
-      The user asks: "can you visualise the average request duration for opbeans-go over the last 7 days?"
-      You recall: {
-        "queries": [
-          "APM service,
-          "lens function usage",
-          "get_apm_timeseries function usage"    
-        ],
-        "categories": [
-          "lens",
-          "apm"
-        ]
-      }`,
```

Comment:
Do we no longer need to provide examples to the LLM about how to use the `context` function? (I see we already define the function arguments in `parameters`  so perhaps it was just leftover from before we used that?)